backend/src/bundles/auth/auth.controller.ts

```diff
@@ -140,6 +152,42 @@ class AuthController extends Controller {
             payload: token,
         };
     }
+
+    /**
+     * @swagger
+     * /auth/authenticated-user:
+     *    get:
+     *      description: Load authenticated user data
+     *      parameters:
+     *        - in: query
+     *          name: token
+     *          schema:
+     *            type: integer
+     *      responses:
+     *        200:
+     *          description: Successful operation
+     *          content:
+     *            application/json:
+     *              schema:
+     *                type: object
+     *                properties:
+     *                  message:
+     *                    type: object
+     *                    $ref: '#/components/schemas/User'
+     */
+
+    private async loadUser(
+        options: ApiHandlerOptions<{
+            query: UserLoadRequestDto;
+        }>,
+    ): Promise<ApiHandlerResponse> {
+        const { token } = options.query;
+        const user = await this.authService.getUserById(token);
```

Comment:
you want to get used by id, but you pass a token.

getUserById should be a user service.
you can create an additional method to get users by token

Reply:
Resolved in [99dba03](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/91/commits/99dba03339f6d85ca8f1515024f336e5f476657a)